eis_toolkit/vector_processing/simple_idw.py

```diff
@@ -0,0 +1,105 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+from beartype import beartype
+from beartype.typing import Optional, Tuple, Union
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+
+
+@beartype
+def _simple_idw(
+    geodataframe: gpd.GeoDataFrame,
+    target_column: str,
+    resolution: Tuple[Number, Number],
+    power: Number,
+    extent: Union[Tuple[Number, Number, Number, Number], None],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+
+    if geodataframe.shape[0] == 0:
+        raise EmptyDataFrameException("Expected geodataframe to contain geometries.")
+
+    if target_column not in geodataframe.columns:
+        raise InvalidParameterValueException(
+            f"Expected value_column ({target_column}) to be contained in geodataframe columns."
+        )
```

Comment:
Maybe move exceptions into the outer function? And then add them into the docstring.